### PR TITLE
Port lgsvl_interface to support planning and control with the SVL simulator

### PR DESCRIPTION
**Pull request by maxime-clem**
## Description

Add the `lgsvl_interface` and its dependencies.
`lgsvl_interface` is used to:
- convert the controllers output to the SVL simulator inputs.
- send information published by the simulator (gear state, steering value, odometry, velocity, etc) to Autoware.

## Review Procedure

You can test the `lgsvl_interface` by following these steps:
- Launch a simulation in the SVL simulator.
- Run the ROS2 bridge.
- Run the `lgsvl_interface`: `ros2 launch lgsvl_interface lgsvl.launch.py`.
- Manually send some control command `ros2 topic pub /vehicle/raw_command autoware_auto_vehicle_msgs/msg/RawControlCommand "{stamp: {sec: 0, nanosec: 0}, throttle: 1.0, brake: 0.0, front_steer: -40, rear_steer: 0.0}"`.

:car:  You can also run Autoware and test the planning and control components by using the following branch of `autoware.proj`:
https://github.com/tier4/autoware.proj/tree/svl-simulator

## Pre-Review Checklist for the PR Author

**PR Author should check the checkboxes below when creating the PR.**

- [x] Read [pull request guidelines][pull-request-guidelines]
- [x] Code follows [coding guidelines][coding-guidelines]
- [x] Assign PR to reviewer

## Checklist for the PR Reviewer

**Reviewers should check the checkboxes below before approval.**

- [ ] Commits are properly organized and messages are according to the guideline
- [ ] Code follows [coding guidelines][coding-guidelines]
- [ ] (Optional) Unit tests have been written for new behavior
- [ ] PR title describes the changes

## Post-Review Checklist for the PR Author

**PR Author should check the checkboxes below before merging.**

- [ ] All open points are addressed and tracked via issues or tickets
- [ ] Write [release notes][release-notes]

## CI Checks

- **Build and test for PR / build-and-test-pr**: Required to pass before the merge.
- **Build and test for PR / clang-tidy-pr**: NOT required to pass before the merge. It is up to the reviewer(s). Found false positives? See the [guidelines][clang-tidy-guidelines].
- **Check spelling**: NOT required to pass before the merge. It is up to the reviewer(s). See [here][spell-check-dict] if you want to add some words to the spell check dictionary.

[clang-tidy-guidelines]: https://tier4.github.io/autoware.proj/tree/main/developer_guide/ClangTidyGuideline/
[coding-guidelines]: https://tier4.atlassian.net/wiki/spaces/AIP/pages/1194394777/T4
[pull-request-guidelines]: https://tier4.github.io/autoware.proj/tree/main/developer_guide/PullRequestGuideline/
[release-notes]: https://tier4.atlassian.net/wiki/spaces/AIP/pages/563774416
[spell-check-dict]: https://github.com/tier4/autoware-spell-check-dict#how-to-contribute

### CODE REVIEW

File: vehicle/lgsvl_interface/param/lgsvl.param.yaml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+      use_nav_odometry_topic: True
+      throttle:
+        domain: [0.0, 3.0]
+        range: [0.0, 100.0]
+      brake:
+        domain: [-3.0, 0.0]
+        range: [100.0, 0.0]
+      steer:
+        domain: [-0.331, 0.331]
+        range: [-100.0, 100.0]
```
Comment (maxime-clem) [inline vehicle/lgsvl_interface/param/lgsvl.param.yaml:18]: @TakaHoribe-san, you can see here the mappings that can be configured for the interface.
Complicated mappings are possible for example:
```yaml
        domain: [0.0, 1.0, 1.5, 2.0, 4.0]
        range: [0.0, 10.0, 20.0, 50.0, 100.0]
```

**Comment by TakaHoribe** [comment]
@maxime-clem in https://github.com/tier4/AutowareArchitectureProposal.iv/pull/677/commits/376964e414d0da0e5d10d0b96860701d840db4bb, do you create pub/sub in the lgsvl_interface? If it is just for renaming, using a [relay](https://github.com/tier4/topic_tools/blob/ac3af33/src/relay.cpp) is the most common way, i think.

You can see the usage in the following links:

https://github.com/tier4/AutowareArchitectureProposal_launcher/blob/use-autoware-auto-msgs/planning_launch/launch/scenario_planning/lane_driving/motion_planning/motion_planning.launch.xml/#L29-L33

https://github.com/tier4/AutowareArchitectureProposal_launcher/blob/use-autoware-auto-msgs/planning_launch/launch/scenario_planning/lane_driving/motion_planning/motion_planning.launch.py/#L108-L121

File: vehicle/lgsvl_interface/param/lgsvl.param.yaml (hunk 1 above)
Comment (TakaHoribe) [reply vehicle/lgsvl_interface/param/lgsvl.param.yaml:18]: Sorry, I missed your coment. Thank you for the information!

Comment (TakaHoribe) [reply vehicle/lgsvl_interface/param/lgsvl.param.yaml:18]: I'm not sure how the LGSVL mapping is complicated, but this mapping usually depends on its velocity.

**Comment by maxime-clem** [comment]
> @maxime-clem in [376964e](https://github.com/tier4/AutowareArchitectureProposal.iv/commit/376964e414d0da0e5d10d0b96860701d840db4bb), do you create pub/sub in the lgsvl_interface? If it is just for renaming, using a [relay](https://github.com/tier4/topic_tools/blob/ac3af33/src/relay.cpp) is the most common way, i think.

Thank you, I did not know about this fork of `topic_tools`. This is great !
I updated the code 
a390c5d and the launch file now uses relay for the clock and odometry (https://github.com/tier4/AutowareArchitectureProposal_launcher/commit/fb00e1b38ab9580cb7bc44027d73a63eb1875977).

**Comment by wep21** [comment]
@TakaHoribe cc @maxime-clem Different formats cause many lint errors. How did you avoid these errors when you ported other .auto codes?

**Comment by TakaHoribe** [comment]
@wep21 we just ignored the test and linter. 
see https://github.com/tier4/AutowareArchitectureProposal.iv/blob/81070acc6367c57978fc43e95cabf2b6dfb6edb4/.pre-commit-config.yaml#L97``

**Comment by wep21** [comment]
@maxime-clem Could you fix or ignore test errrors?

**Pull request closed without merging**